### ISSUE TITLE
Bug: Parsing error: Unexpected token 0

### ISSUE THREAD

**Issue by edhenderson** [eslint/eslint#16140]
### Environment

```
Node version: v16.5.0
npm version: v7.19.1
Local ESLint version: v8.3.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.4.0
```

```
module.exports = {
  env: {
    browser: true,
    node: true
  },
  plugins: ['njk'],
  extends: 'eslint:recommended',
  parserOptions: {
    ecmaVersion: 13
  },
  rules: {
    indent: ['error', 2],
    'linebreak-style': ['error', 'unix'],
    quotes: ['error', 'single', { allowTemplateLiterals: true }],
    semi: ['error', 'always'],
    'no-unused-vars': 'off'
  }
};
```

### What parser are you using?

Default (Espree)

### What did you do?

Line in question:
```js
const { blocks = [] } = rawPageItem.fields;
```

Red error line appears in VSCode as per below:

<img width="537" alt="Screenshot 2022-07-18 at 08 05 18" src="https://user-images.githubusercontent.com/1378582/179459974-3f7fcefb-9a4c-4045-b0a5-2433b9fd8360.png">

### What did you expect to happen?

This error to not be flagged. It happens elsewhere but not consistently. However there are other places in my code, this does not happen.

### What actually happened?

<img width="378" alt="Screenshot 2022-07-18 at 08 06 40" src="https://user-images.githubusercontent.com/1378582/179460157-528a52b5-f1f2-4c2a-b96e-28aec6bbeaae.png">

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

First time reporting this, so please let me know if there is anything else I can do to help. Thanks.

**Comment by coderaiser** [comment]
What is `njk` plugin? There is no such a plugin as `eslint-plugin-njk` in npm. Maybe it’s related.

**Comment by edhenderson** [comment]
@coderaiser Thanks for the prompt response. This isn't my project so let me tell you what I can find:

From `package.json`:

```
"eslint-plugin-njk": "file:eslint-plugin-njk",
```

Then from it's local `package.json`:

```
{
  "name": "eslint-plugin-njk",
  "version": "0.0.1",
  "description": "A preprocessor to remove nunjuck tags",
  "keywords": [
    "eslint",
    "eslintplugin",
    "eslint-plugin",
    "eslint-nunjuck-parse-fix",
    "eslint-nunjuck-fix"
  ],
  "author": {
    "name": "Ibrahim Rashid",
    "email": "irashid.com@gmail.com"
  },
  "repository": {
    "type": "git",
    "url": "git+git@github.com:ibrahim12/eslint-plugin-exclude-nunjuck-tags.git"
  },
  "main": "lib/index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": { },
  "devDependencies": {
    "eslint": "^2.6.0",
    "mocha": "^2.4.5"
  },
  "engines": {
    "node": ">=4"
  },
  "license": "BSD-3-Clause"
}
```

**Comment by edhenderson** [comment]
https://npm.io/package/eslint-plugin-exclude-nunjuck-tags

**Comment by mdjermanovic** [comment]
> <img alt="Screenshot 2022-07-18 at 08 06 40" width="378" src="https://user-images.githubusercontent.com/1378582/179460157-528a52b5-f1f2-4c2a-b96e-28aec6bbeaae.png">

This does seem to be caused by `eslint-plugin-exclude-nunjucks-tags`, as it [replaces code with `0`](https://github.com/ibrahim12/eslint-plugin-exclude-nunjuck-tags/blob/67967324a69dfcccaa7e9891d46c420cae169efd/lib/index.js#L4).

@edhenderson can you remove `plugins: ['njk']` from your config file and try linting the same code again?

**Comment by edhenderson** [comment]
@mdjermanovic I did some digging today and removed plugin, that does indeed fix it. My thanks and apologies for all the help. Very much appreciated.
